test/cptree.hs

```diff
@@ -0,0 +1,28 @@
+{-# LANGUAGE OverloadedStrings #-}
+
+import Turtle
+import Filesystem.Path.CurrentOS ()
+import System.IO.Temp (withSystemTempDirectory)
+import Control.Monad (unless)
+
+check :: String -> Bool-> IO ()
+check errorMessage successs = unless successs $ error errorMessage
+
+-- This test fails by hanging
+main :: IO ()
+main = withSystemTempDirectory "tempDir" (runTest . fromString)
```

Comment:
I saw Turtle.tempdir, but I prefer withSystemTempDirectory because it will use the correct system temp directory if it is not "/tmp":
```
On Unix, getTemporaryDirectory returns the value of the TMPDIR environment variable or "/tmp" if the variable isn't defined. On Windows, the function checks for the existence of environment variables in the following order and uses the first path found:

    TMP environment variable.
    TEMP environment variable.
    USERPROFILE environment variable.
    The Windows directory
```
https://hackage.haskell.org/package/directory-1.2.2.0/docs/System-Directory.html#v:getTemporaryDirectory